### PR TITLE
Proofpoint TAP v2 - Added a validation to the start time

### PR DESCRIPTION
**Pull request by RotemAmit**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
relates: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/XSUP-36905)

## Description
* Since the API states that "the maximum time into the past that can be queried is 7 days", I added a validation to enforce it, instead of getting an exception from the API.
* Updated the fetch-incidents to use `dateparser.parse()` instead of `parse_date_range()`, because it is deprecated.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 1]
```diff
@@ -537,6 +537,25 @@ def get_events_command(client, args):
...
 
+def validate_start_query_time(start_query_time: str):
+    """
+        validate that the start time is less than 7 days ago
+        Args:
+            start_query_time(str) - the start date time that needs to be validated.
+        Returns:
+            A valid datetime for the start_query_time
+        """
+    demisto.debug(f'In validate_start_query_time {start_query_time=}')
```
Comment (MLainer1) [inline Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:548]:
```suggestion
```

**Review by MLainer1 — APPROVED** [review]
Nice! 🆒

File: Packs/ProofpointTAP/ReleaseNotes/1_2_12.md [hunk 2]
```diff
@@ -0,0 +1,4 @@
+#### Integrations
+
+##### Proofpoint TAP v2
+Fixed an issue where too old start date could be sent in the request to the API in the fetch-incidents command.
```
Comment (eyalpalo) [inline Packs/ProofpointTAP/ReleaseNotes/1_2_12.md:4]:
```suggestion
Fixed an issue where excessively old start dates could be sent API in the fetch-incidents command.
```

**Review by eyalpalo — CHANGES_REQUESTED** [review]

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 3]
```diff
@@ -537,6 +537,24 @@ def get_events_command(client, args):
...
 
+def validate_start_query_time(start_query_time: str):
+    """
+        validate that the start time is less than 7 days ago
+        Args:
+            start_query_time(str) - the start date time that needs to be validated.
+        Returns:
+            A valid datetime for the start_query_time
+        """
+    dt_start_query_time = datetime.strptime(start_query_time, DATE_FORMAT) if start_query_time else get_now() - timedelta(hours=1)
```
[the remark this answers is not in the document]
Comment (RotemAmit) [reply Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:548]: yes

**Review by eyalpalo — APPROVED** [review]
Nice

**Review by MLainer1 — COMMENTED** [review]
😎

File: Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py [hunk 4]
```diff
@@ -562,6 +579,7 @@ def fetch_incidents(
     # Handle first time fetch, fetch incidents retroactively
     if not start_query_time:
         start_query_time, _ = parse_date_range(first_fetch_time, date_format=DATE_FORMAT, utc=True)
+
```
Comment (eyalpalo) [inline Packs/ProofpointTAP/Integrations/ProofpointTAP_v2/ProofpointTAP_v2.py:582]: Please remove

**Review by eyalpalo — APPROVED** [review]

**Pull request merged**